src/Halogen/Menu/Component.purs

```diff
@@ -1,133 +1,107 @@
--- | available
--- | [here](https://github.com/beckyconning/color-editor/blob/master/stylesheet.css).
--- |
--- | If you are providing keyboard shortcut labels to be rendered we recommend
--- | using the flexible box model on the submenu item's anchor and a fifty
--- | percent width for the spans inside it. If you can't use the flexible box
--- | model we reccomend you define a fixed width for your submenus.
-menuComponent :: forall a g. Functor g => H.Component (MenuP a g) (MenuQueryP a) g
-menuComponent = H.parentComponent { render, eval, peek: Just (peek <<< H.runChildF) }
-
-render :: forall a g. Menu a -> HTML a g
-render menu =
-  HH.ul_ $ mapWithIndex (renderSubmenu menu) menu.submenus
-
-renderSubmenu :: forall a g. Menu a -> Int -> MenuItem a -> HTML a g
-renderSubmenu menu index menuSubmenu
-  | menu.chosen == Just index = renderChosenSubmenu index menuSubmenu
-  | otherwise = renderHiddenSubmenu index menuSubmenu
-
-renderChosenSubmenu :: forall a g. Int -> MenuItem a -> HTML a g
-renderChosenSubmenu index menuSubmenu =
-  HH.li_
-    [ renderAnchor DismissSubmenu menuSubmenu.label
-    , HH.slot (SubmenuSlotAddress index) \_ ->
-        { component: submenuComponent
-        , initialState: menuSubmenu.submenu
-        }
-    ]
-
-renderHiddenSubmenu :: forall a g. Int -> MenuItem a -> HTML a g
-renderHiddenSubmenu index menuSubmenu =
-  HH.li_ [ HH.div_ [ renderAnchor (SelectSubmenu index) menuSubmenu.label ] ]
-
-renderAnchor :: forall a g. H.Action (MenuQuery a) -> String -> HTML a g
-renderAnchor a label =
-  HH.a
-    [ HE.onClick (\_ -> HEH.preventDefault *> HEH.stopPropagation $> Just (H.action a)) ]
-    [ HH.text $ label ]
-
-eval :: forall a g. MenuQuery a ~> DSL a g
-eval (SelectSubmenu index next) = H.modify (_ { chosen = Just index }) $> next
-eval (DismissSubmenu next) = H.modify (_ { chosen = Nothing }) $> next
-eval (SetMenu menu next) = H.set menu $> next
-
-peek :: forall a g x. SubmenuQuery a x -> DSL a g Unit
-peek (SelectSubmenuItem _ _) = H.modify (_ { chosen = Nothing }) *> pure unit
+import Halogen.Menu.Submenu.Component as SUB
+
+data Query a next
+  = Set (State a) next
+  | SelectSubmenu Int DET.MouseEvent next
+  | DismissSubmenu DET.MouseEvent next
+  | HandleSubmenu (SUB.Message a) next
+
+type State a =
+  { chosen ∷ Maybe Int
+  , submenus ∷ Array (MenuItem a)
+  }
+
+type MenuItem a = { label ∷ String, submenu ∷ SUB.State a }
+
+data Message a = Selected a
+
+type HTML a m = H.ParentHTML (Query a) (SUB.Query a) Int m
+type DSL a m = H.ParentDSL (State a) (Query a) (SUB.Query a) Int (Message a) m
+type Input a = Array { label ∷ String, submenu ∷ SUB.State a }
+
+component
+  ∷ ∀ m a r
+  . (MonadAff (dom ∷ DOM|r) m)
+  ⇒ H.Component HH.HTML (Query a) (Input a) (Message a) m
+component = H.parentComponent
+  { initialState: \submenus → { chosen: Nothing, submenus }
+  , receiver: \submenus → Just $ H.action $ Set { chosen: Nothing, submenus }
```

Comment:
If you are resetting `chosen` to Nothing, this means every time a parent renders, the selection will be cleared.